src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -389,7 +389,8 @@ Handle java_lang_String::create_from_symbol(Symbol* symbol, TRAPS) {
   }
 
 #ifdef ASSERT
-  {
+  // This check is too strict when the symbol does not contain valid UTF8.
+  if (UTF8::is_legal_utf8((const unsigned char*)utf8_str, utf8_len, false)) {
```

Comment:
But a successfully created Symbol should only ever be valid UTF8. ??

Reply:
In theory, yes. However, there was a single test which somehow generated corrupted Symbols. I did not investigate this much further, and instead relaxed the check to look the same as in `java_lang_String::create_from_str`.

Reply:
That seems a bug as it should not be possible. Please dig up the failing test case so I/we can examine it further - thanks.